docs/api.md

```diff
@@ -1147,20 +1236,216 @@ Only `DELETE` a `Form` if you are sure you will never need it again. If your goa
+
++ Response 200 (application/xml)
+    + Body
+
+            (binary data)
+
++ Response 403 (application/json)
+    + Attributes (Error 403)
+
+#### Listing expected Form Attachments [GET /v1/projects/{projectId}/forms/{xmlFormId}/draft/attachments]
+
+As mentioned above, the list of expected form attachments is determined at form creation time, from the XForms definition. This endpoint allows you to fetch that list of expected files, and will tell you whether the server is in possession of each file or not.
+
++ Response 200 (application/json)
+    + Attributes (array[Form Attachment])
+
++ Response 403 (application/json)
+    + Attributes (Error 403)
+
+#### Uploading a Form Attachment [POST /v1/projects/{projectId}/forms/{xmlFormId}/draft/attachments/{filename}]
+
+To upload a binary to an expected file slot, `POST` the binary to its endpoint. Supply a `Content-Type` MIME-type header if you have one.
+
++ Parameters
+    + xmlFormId: `simple` (string, required) - The `xmlFormId` of the Form being referenced.
+
++ Request (*/*)
+    + Body
+
+            (binary data)
+
++ Response 200 (application/json)
+    + Attributes (Success)
+
++ Response 403 (application/json)
+    + Attributes (Error 403)
+
+#### Downloading a Form Attachment [GET /v1/projects/{projectId}/forms/{xmlFormId}/draft/attachments/{filename}]
+
+To download a single file, use this endpoint. The appropriate `Content-Disposition` (attachment with a filename) and `Content-Type` (based on the type supplied at upload time) will be given.
+
++ Parameters
+    + xmlFormId: `simple` (string, required) - The `xmlFormId` of the Form being referenced.
+
++ Response 200
+    + Headers
+
+            Content-Type: {the MIME type of the attachment file itself}
+            Content-Disposition: attachment; filename={the file's name}
+
+    + Body
+
+            (binary data)
+
++ Response 403 (application/json)
+    + Attributes (Error 403)
+
+#### Clearing a Form Attachment [DELETE /v1/projects/{projectId}/forms/{xmlFormId}/draft/attachments/{filename}]
+
+Because Form Attachments are completely determined by the XForms definition of the form itself, there is no direct way to entirely remove a Form Attachment entry from the list, only to clear its uploaded content. Thus, when you issue a `DELETE` to the attachment's endpoint, that is what happens.
 
 + Parameters
     + xmlFormId: `simple` (string, required) - The `xmlFormId` of the Form being referenced.
+
++ Response 200 (application/json)
+    + Attributes (Success)
+
++ Response 403 (application/json)
+    + Attributes (Error 403)
+
+#### Getting Draft Form Schema Fields [GET /v1/projects/{projectId}/forms/{xmlFormId}/draft/fields{?odata}]
+
+Identical to the [same request](/reference/forms-and-submissions/'-individual-form/retrieving-form-schema-fields) for the root Form, but will return the fields related to the current Draft version.
```

Comment:
How about "published Form" or "active Form" instead of "root Form"? Or "root Form endpoint"?